server/src/main/java/io/crate/expression/symbol/Symbols.java

```diff
@@ -268,7 +268,7 @@ public Void visitReference(Reference ref, Void context) {
                     } else if (ref instanceof VoidReference
                                && t instanceof ScopedSymbol tScopedSymbol
                                && tScopedSymbol.relation().equals(ref.ident().tableIdent())
-                               && tScopedSymbol.column().equals(root)) {
+                               && (tScopedSymbol.column().equals(root) || tScopedSymbol.column().equals(ref.column()))) {
```

Comment:
The existing logic allows `obj_dynamic['u']`(needle) to match `obj_dynamic`(haystack) from `select obj_dynamic['u'] from (select obj_dynamic from t) t2;` 
We also need to allow `obj_dynamic['u']` to match `obj_dynamic['u']` from `select obj_dynamic['u'] from (select obj_dynamic['u'] from t) t2;`